src/app_service/service/wallet_account/service.nim

```diff
@@ -857,3 +859,13 @@ QtObject:
         if self.walletAccounts[address].tokens.len > 0:
           return true
     return false
+
+  proc hasMarketCache*(self: Service): bool =
+    withLock self.walletAccountsLock:
```

Comment:
In some point we will need to remove locking for `walletAccounts`, cause it's useless, execution of all services is happening in the main thread.